test.js

```diff
@@ -50,6 +50,30 @@ test('register', function (t) {
     })
   })
 
+  t.test('secret as a promise', function (t) {
+    t.plan(1)
+    const fastify = Fastify()
+    fastify.register(jwt, {
+      secret: async function () {
+        return await Promise.resolve('supersecret')
```

Comment:
As a general rule, never do: `return await promise`. The `await` is not needed at all, it will be done automatically.

Reply:
Thanks for the tip. I removed all async/await